### PR TITLE
Changed to ant design radio buttons

### PR DESCRIPTION
**Pull request by daemon-reconfig**
## Description

this fixes: [#912 ](https://github.com/PolicyEngine/policyengine-app/issues/912)

## Changes

Changed to ant design radio buttons 

## Screenshots

Screenshot 2024-04-13 at 2 15 37 AM

### CODE REVIEW

**Comment by anth-volk** [comment]
I'm noticing that the border radius on this isn't quite squared off. Could you set it to 0? When we migrate to Ant Design v5, this will become much rounder, too much for our design.

**Comment by anth-volk** [comment]
Could you run `make format` and push the results? We do require linting before merging in order to make for better code quality.

**Comment by daemon-reconfig** [comment]
sure i will do so

**Comment by daemon-reconfig** [comment]
Screenshot 2024-04-13 at 10 34 29 AM

here fixed

**Comment by daemon-reconfig** [comment]
Screenshot 2024-04-13 at 10 59 36 AM

it gives this while linting

**Comment by MaxGhenis** [comment]
Try `pip install black` first (see https://github.com/PolicyEngine/policyengine-app/issues/1602)

**Comment by daemon-reconfig** [comment]
> Try `pip install black` first (see #1602)

thanks!

**Comment by daemon-reconfig** [comment]
@anth-volk could you review this PR and merge it if it suits ?

**Comment by anth-volk** [comment]
I will try to do so today, but am not sure if I'll be able to. It didn't appear that you'd run the linter, which is why I held off, but the code does pass all tests, so perhaps you had?

**Comment by daemon-reconfig** [comment]
yes had linted the code

File: package.json [hunk 1]
```diff
@@ -14,8 +14,9 @@
     "@uiw/codemirror-theme-github": "^4.21.25",
     "@uiw/react-codemirror": "^4.21.25",
     "@uiw/react-textarea-code-editor": "^3.0.2",
-    "antd": ">4, <5",
+    "antd": "^4.24.16",
     "axios": "^1.6.8",
+    "black": "^0.3.0",
```
Comment (anth-volk) [inline package.json:19]:
```suggestion
```

Comment (anth-volk) [reply package.json:19]: This package is different from the Python `black` linting package and should be uninstalled via `npm uninstall black` before recommitting

**Review by anth-volk — CHANGES_REQUESTED** [review]
Thanks for this, @daemon-reconfig. Just a couple of things:

1. If you run the code, you'll notice that the borders between the buttons are poorly defined, since they're a shade of gray. This is probably because of how Ant Design creates these borders. Could you instead iterate over the labels and generate individual `Radio.Button` elements to sit within one `Radio.Group`, and have any non-selected buttons have their own `style={{}}` prop that contains `border: "1px solid style.colors.WHITE"`?
2. This component currently doesn't manage styling statefully, instead always just making the background of the first element in the list blue and keeping the others gray. Could you use a standard state handling structure within `Radio.Group`'s `onChange` to save something (perhaps an index) to a stateful variable, then conditionally apply a background color to each `Radio.Button` within the `Radio.Group`? This will probably require a function that looks something like the following:

```
function handleChange(may need args here, depending on how Ant Design passes events) {

  Check if PolicyEngine's RadioButton's onChange prop is a function

  If so, run that function

  No matter what, update state to make this a controlled component
}
```

I've yet to actually test to make sure values are passed properly, but they appear to be.

Thanks again.

**Comment by daemon-reconfig** [comment]
sure I will do the following changes.

**Comment by daemon-reconfig** [comment]
if I uninstall that black I get error while linting

**Comment by daemon-reconfig** [comment]
@anth-volk could you please elaborate what you want me to add in it

**Comment by daemon-reconfig** [comment]
should I map radio.buttons? and 0th index will be the selected one the border of non selected should be changed to white
is this what you are talking about?
or I can change the border color of the non selected button to white

**Comment by anth-volk** [comment]
The issue with `black` is partly on us, and it's being fixed in #1623. The `black` that you installed is a JavaScript package, but what contributors actually need is the Python package `black`, so if you could run `make install`, then `pip install -U black` for the time being, that should fix your issue. If that fails, please let me know.

**Comment by anth-volk** [comment]
And yes, the borders should be white for all non-selected buttons, but what you have now for the selected button works. You should probably place `Radio.Group` into the main return statement, since you only need one of those, then map `Radio.Button`, but adding event change handling, akin to building a standard fully controlled React component. I could go more into detail if you're unfamiliar with this.

**Comment by daemon-reconfig** [comment]
I had done pip install black when I installed it @MaxGhenis told me to.
an sure I will add an event handler for the buttons but I didn't get the thing about radio.button

**Comment by daemon-reconfig** [comment]
1. I am so sorry about it I did run `npm i black`
I will uninstall it.

2. I do know why we need states but I didn’t get why we should add `Radio.Button` as in the given reference there was no mention of it
https://ant.design/components/radio
But since you asked me to I will add them to it
what I did get is
• Adding a `Radio.Button` so that the first button would be selected which will have normal border- blue given by ant design itself 
• the second button which would be unselected should have white borders
• above two points should be achieved using states which will change the colors of the first button if it is unselected, i.e, second button was selected

my opinion: since we would not just be targeting two buttons using `Radio.Button` might not do it although we can map an array which will give us `Radio.Button`.

I deeply apologise if I caused any misunderstanding. 
I am thankful for your helpful messages and you have helped me learn a lot of things.

**Comment by anth-volk** [comment]
1. Not a problem at all
2. Ant Design's documentation, unfortunately, isn't always amazing. I'm thinking `Radio.Button` would be better because with just the group, it seems a bit more difficult to manage state and to conditionally format the buttons. If you can get it to work with just `Radio.Group`, please feel free, that was just my read of the documentation.

**Comment by anth-volk** [comment]
Also, in the future, if you're still working on an issue, please feel free to convert it to "draft" by clicking "Still in progress? Convert to draft" in the top right. Typically, if you close a PR, it looks like you no longer want to work on the issue.

**Comment by daemon-reconfig** [comment]
sure I will add it to draft. 
also I think we can make do with `Radio.Group` and I will start working on implementing `Radio.Button` in it. as it may take time since I will have to research a bit into it.
or we can either go with more time taken but the end result would be getting a `Radio.Button`

**Comment by anth-volk** [comment]
We're open to however you'd like to implement it, whichever way you think will work best.

**Comment by daemon-reconfig** [comment]
sure will open a PR for it on Sunday. I have an exam tomorrow

**Pull request closed without merging**